### ISSUE TITLE
The device has been disconnecting  error while trying to send data to hub , could anyone tell  me what is the main reason?

### ISSUE THREAD

**Issue by chinnaece** [Azure/azure-iot-sdk-c#776]
I am using MQTT as a protocol, i just trying to change DEFAULT_CONNACK_TIMEOUT value  in the iothubtransport_mqtt_commn.c  . 

 this is function :
(void)IoTHubDeviceClient_LL_SetConnectionStatusCallback(device_ll_handle, connection_status_callback, NULL);

static void connection_status_callback(IOTHUB_CLIENT_CONNECTION_STATUS result, IOTHUB_CLIENT_CONNECTION_STATUS_REASON reason, void* user_context)
{
    (void)reason;
    (void)user_context;
    // This sample DOES NOT take into consideration network outages.
    if (result == IOTHUB_CLIENT_CONNECTION_AUTHENTICATED)
    {
        (void)printf("The device client is connected to iothub\r\n");
    }
    else
    {
        (void)printf("The device client has been disconnected\r\n");
    }
}

**Comment by jebrando** [comment]
@chinnaece The way you would set the DEFAULT_CONNACK_TIMEOUT value is this:

```C
int conn_ack_seconds = 20;
IoTHubDeviceClient_LL_SetOption(iothub_ll_handle, OPTION_CONNECTION_TIMEOUT, &conn_ack_seconds);
```

You should set this right after you call IoTHubDeviceClient_LL_CreateFromConnectionString.

**Comment by az-iot-builder-01** [comment]
@chinnaece, @jebrando, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)

**Comment by chinnaece** [comment]
i would like send data for every one minute , even i set  **int conn_ack_seconds = 3*60;**  , DEFAULT_CONNACK_TIMEOUT as 4*60 , still the device has been disconnecting.

**Comment by chinnaece** [comment]
> @chinnaece The way you would set the DEFAULT_CONNACK_TIMEOUT value is this:
> 
> ```c
> int conn_ack_seconds = 20;
> IoTHubDeviceClient_LL_SetOption(iothub_ll_handle, OPTION_CONNECTION_TIMEOUT, &conn_ack_seconds);
> ```
> 
> You should set this right after you call IoTHubDeviceClient_LL_CreateFromConnectionString.

Still not working

**Comment by chinnaece** [comment]
the problem solved explained in #782
